ironfish/src/wallet/wallet.ts

```diff
@@ -939,6 +940,26 @@ export class Wallet {
         })
       }
 
+      if (options.markSpent) {
+        await this.walletDb.db.transaction(async (tx) => {
+          for (const spend of raw.spends) {
+            const decryptedNote = await options.account.getDecryptedNote(spend.note.hash())
+
+            Assert.isNotUndefined(decryptedNote)
+
+            await this.walletDb.saveDecryptedNote(
```

Comment:
Should this also be calling this?

```ts
await this.walletDb.deleteUnspentNoteHash(this, spentNoteHash, spentNote, tx)
```

otherwise next time `createSpendsForAsset` calls `getUnspentNotes`, I think it'll still return the note

Reply:
I'd probably expect something like this code block, but I haven't looked enough to see if all of these are necessary: https://github.com/iron-fish/ironfish/blob/5f68460f860d5447f9152d218effa466b58d798b/ironfish/src/wallet/account.ts#L495-L512

Reply:
Yes! We need to update the index, too.

I was also looking at your comment here: https://github.com/iron-fish/ironfish/pull/3721/files#r1164678053, and I think that that might be a better option than adding a permanent parameter 🤔

Reply:
> I'd probably expect something like this code block, but I haven't looked enough to see if all of these are necessary:
> 
> https://github.com/iron-fish/ironfish/blob/5f68460f860d5447f9152d218effa466b58d798b/ironfish/src/wallet/account.ts#L495-L512

We can't update the `nullifierToTransactionHash` index here because we don't have the transaction hash until the transaction is posted

Reply:
I'm not certain what people will want to do long-term -- They almost definitely will want to specify which notes will go into a transaction though, but that's probably a bit trickier to get ready.

I could see a use for this, but it's a bit dangerous in that there isn't a way to un-spend a note without adding a corresponding transaction and waiting for it to expire.

Adding a temporary hack would probably be the least commitment, and I'm okay with that since it'd just live in the `airdrop` branch